src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,39 @@
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(country);
+            transaction.commit();
+            return country;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add country :" + country.getName(), e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
```

Comment:
Remember to use try-with-resources when you are dealing with AutoCloseable resources such as Session. This is a code quality issue. It will help to ensure that the Session is always closed, even if an exception is thrown.